src/Button/Button.tsx

```diff
@@ -2,9 +2,10 @@ import styled from 'styled-components'
 import sx from '../sx'
 import {get} from '../constants'
 import theme from '../theme'
-import ButtonBase, {systemStyles} from './ButtonBase'
+import ButtonBase, {ButtonBaseProps, systemStyles} from './ButtonBase'
+import {ComponentProps} from '../utils/types'
 
-const Button = styled(ButtonBase)`
+const Button = styled(ButtonBase)<ButtonBaseProps>`
```

Comment:
I _think_ styled can infer the props accepted by `ButtonBase`. It's worth testing out though.
```suggestion
const Button = styled(ButtonBase)`
```

Reply:
Thanks, it looks like you’re right! Out of curiosity—when is an explicit type necessary? I noticed one is provided in [Avatar](https://github.com/primer/components/blob/b894ead466822e42b8f659f37a37ec3c71876506/src/Avatar.tsx#L22), [Grid](https://github.com/primer/components/blob/b894ead466822e42b8f659f37a37ec3c71876506/src/Grid.tsx#L7), and [StyledOcticon](https://github.com/primer/components/blob/b894ead466822e42b8f659f37a37ec3c71876506/src/StyledOcticon.tsx#L16), so I thought it was a Primer code style convention and tried to follow it here. I’m now realizing there might be a specific need for it in those cases?

Reply:
Removed `<ButtonBaseProps>` from `Button` and similar in https://github.com/primer/components/pull/1017/commits/1d60b555f0dbd263501ea3eb46e4be2542c32de2

Reply:
Note: An explicit type was restored in https://github.com/primer/components/pull/1017/commits/7f8997e9748aaadd949c420b46915396508e8532, which added `ButtonSystemProps`

Reply:
Note: The explicit type restored in https://github.com/primer/components/commit/7f8997e9748aaadd949c420b46915396508e8532 was amended in https://github.com/primer/components/commit/80317dac298e3f442aaa535bad19fcf18b808089, because `sx` was apparently not `compose`able, so I applied it separately.